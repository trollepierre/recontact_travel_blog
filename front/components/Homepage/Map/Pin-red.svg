<svg id="Ebene_1" data-name="Ebene 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 400 400">
  <style>.pin{fill: #F48024} .center{fill:#07C}</style>
  <path
    class='pin' d="M200,74a80.33,80.33,0,0,0-80.32,80.33,32.43,32.43,0,0,0,.1,3.82,80,80,0,0,0,2.7,17.33C128.35,201.24,146.15,250.84,200,326c53.45-74.64,71.39-124.08,77.36-150a79.86,79.86,0,0,0,2.91-18.81c.05-1,.06-1.75.06-2.22,0-.22,0-.43,0-.64A80.34,80.34,0,0,0,200,74Z"/>
  <circle id="Ellipse_2" data-name="Ellipse 2" class="center" cx="198.45" cy="154.82" r="23.38"/>
</svg>
