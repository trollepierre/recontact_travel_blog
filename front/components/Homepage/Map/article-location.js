export const articleLocation = [
  {
    // https://www.gps-longitude-latitude.net/
    location: { lng: 3.518332, lat: 50.357113 },
    frTitle: 'Cols verts Valenciennes',
    enTitle: '75 bis rue Jean Jaurès',
    url: 'http://www.lescolsverts.com/collectifs/valenciennes',
  },
  {
    location: { lng: -1.6586876, lat: 48.0894659 },
    frTitle: 'Cols verts Rennes',
    enTitle: 'Le potager des cultures',
    url: 'http://www.lescolsverts.com/collectifs/rennes',
  },
  {
    location: { lng: 2.1700918, lat: 43.94202 },
    frTitle: 'Cols verts Albi',
    enTitle: 'Ferme du Gô, Urba Terra',
    url: 'http://www.lescolsverts.com/collectifs/albi',
  },
  {
    location: { lng: 3.876716, lat: 43.610769 },
    frTitle: 'Cols verts Montpellier',
    enTitle: 'Bassin de Thau - quartier de Frontignan',
    url: 'http://www.lescolsverts.com/collectifs/montpellier',
  },
  {
    location: { lng: 7.752111, lat: 48.573405 },
    frTitle: 'Cols verts Strasbourg',
    enTitle: undefined,
    url: 'http://www.lescolsverts.com/collectifs/strasbourg',
  },
  {
    location: { lng: -61.024174, lat: 14.641528 },
    frTitle: 'Ta Nou',
    enTitle: undefined,
    url: 'http://www.lescolsverts.com/collectifs/martinique',
  },
  {
    location: { lng: 11.045721, lat: 35.502446 },
    enTitle: 'Cols verts Tunisie',
    frTitle: 'Mahdia',
    url: 'http://www.lescolsverts.com/collectifs/mahdia',
  },
];
