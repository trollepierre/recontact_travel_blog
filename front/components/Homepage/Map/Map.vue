<template>
  <div id="map"/>
</template>

<script>
  import mapboxgl from 'mapbox-gl'
  import { articleLocations } from './article-location'
  import translationService from '../../../services/services/translations'

  export default {
    data() {
      return {
        selectedMarker: null,
        map: null,
        markers: [],
        articleLocations,
      }
    },
    mounted() {
      /* istanbul ignore file */

      // https://gist.github.com/mornir/9e85e65caba46e55269302e8a134e04e
      // https://www.mapbox.com/install/js/bundler-complete/
      // https://docs.mapbox.com/mapbox-gl-js/example/

      this.map = new mapboxgl.Map({
        accessToken: process.env.NUXT_ENV_MAPBOX_API_TOKEN,
        container: 'map',
        style: 'mapbox://styles/mapbox/streets-v11', // default style // V11 - V9 => to try
        center: window.innerWidth > 1024 ? [30, 0] : [30, 30], // mobile center to Lybia
        zoom: window.innerWidth > 650 ? 1 : 0, // 1 = monde, 2 = europe, 3 = west europe, 4 = france,
      })

      this.markers = this.articleLocations.map(articleLocation => {
        const title = translationService.isFrancophone() ? articleLocation.frTitle : articleLocation.enTitle
        const popup = `<a href="${articleLocation.url}" class="font-bold">${title}</a>`
        // if (articleLocation.description) {
        //   popup += `<p>${articleLocation.description}</p>`
        // }
        // popup += `<p>${articleLocation.address}</p>`
        // popup += `<p>${articleLocation.city}</p>`
        // popup += `<br><p>${articleLocation.remark}</p>`

        const LngLat = [articleLocation.location.lng, articleLocation.location.lat]

        const element = document.createElement('div')
        element.className = `marker ${articleLocation.color}`
        element.addEventListener('click', e => {
          // flyTo sometimes fails to trigger the close event on the popup
          if (this.selectedMarker) {
            this.selectedMarker.classList.remove('hidden')
          }
          this.selectedMarker = e.target
          this.selectedMarker.classList.add('hidden')
          this.map.flyTo({ center: LngLat, speed: 0.5 })
        })

        const popupElement = new mapboxgl.Popup({
          offset: 0,
          anchor: 'bottom',
        }).setHTML(popup)
        popupElement.on('close', () => {
          if (this.selectedMarker) {
            this.selectedMarker.classList.remove('hidden')
          }
        })

        const marker = new mapboxgl.Marker({ element, offset: [0, -10] })
          .setLngLat(LngLat)
          .setPopup(popupElement)
          .addTo(this.map)
        marker.description = title

        return marker
      })
    },
  }
</script>
<style>
#map {
  height: 256px;
  width: 256px;
  margin: auto;
}

.marker {
  background-image: url('Pin.svg');
  background-size: cover;
  width: 40px;
  height: 40px;
  cursor: pointer;
}

.blue {
  background-image: url('Pin-blue.svg');
}

.red {
  background-image: url('Pin-red.svg');
}

.green {
  background-image: url('Pin-green.svg');
}

@media only screen and (min-width: 650px) {
  #map {
    height: 400px;
    width: 600px;
  }
}

@media only screen and (min-width: 1024px) {
  #map {
    height: 600px;
    width: 900px;
  }
}
</style>
