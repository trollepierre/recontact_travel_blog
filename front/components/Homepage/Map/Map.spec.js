import Map from '@/components/Homepage/Map/Map';

jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
  Map: jest.fn(),
  Marker: jest.fn().mockReturnValue({
    setLngLat: jest.fn().mockReturnValue({
      setPopup: jest.fn().mockReturnValue({
        addTo: jest.fn().mockReturnValue({}),
      }),
    }),
  }),
  Popup: jest.fn().mockReturnValue({
    setHTML: jest.fn().mockReturnValue({ on: jest.fn() }),
  }),
}));

describe('Map', () => {
  it('should match snapshot', () => {
    // When
    const wrapper = shallowMount(Map);

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});
