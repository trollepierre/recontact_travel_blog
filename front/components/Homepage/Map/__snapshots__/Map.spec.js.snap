// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map should match snapshot 1`] = `<div id="map"></div>`;
